quickjs.c

```diff
@@ -40007,10 +40021,44 @@ static JSValue js_iterator_from(JSContext *ctx, JSValue this_val,
     return iter;
 }
 
+static JSValue js_create_iterator_helper(JSContext *ctx, JSValue iterator,
+                                         JSIteratorHelperKindEnum kind, JSValue func, int64_t limit);
+
 static JSValue js_iterator_proto_drop(JSContext *ctx, JSValue this_val,
                                       int argc, JSValue *argv)
 {
-    return JS_ThrowInternalError(ctx, "TODO implement Iterator.prototype.drop");
+    JSValue v;
+    double dlimit;
+    int64_t limit;
+    if (!JS_IsObject(this_val))
+        return JS_ThrowTypeError(ctx, "Iterator.prototype.drop called on non-object");
+    v = JS_ToNumber(ctx, argv[0]);
+    if (JS_IsException(v))
+        return JS_EXCEPTION;
+    if (JS_VALUE_IS_NAN(v))
+        goto fail;
+    // Check for Infinity.
+    if (JS_ToFloat64(ctx, &dlimit, v)) {
+        JS_FreeValue(ctx, v);
+        return JS_EXCEPTION;
+    }
```

Comment:
Checking JS_VALUE_IS_NAN first isn't sound because the `JS_ToFloat64` call can also produce NaN. The logic should look like this:
```suggestion
    // Check for Infinity.
    if (JS_ToFloat64(ctx, &dlimit, v)) {
        JS_FreeValue(ctx, v);
        return JS_EXCEPTION;
    }
    if (isnan(dlimit)) {
        JS_FreeValue(ctx, v);
        goto fail;
    }
```